airflow/operators/trigger_dagrun.py

```diff
@@ -185,3 +215,40 @@ def execute(self, context: Context):
                 if state in self.allowed_states:
                     self.log.info("%s finished with allowed state %s", self.trigger_dag_id, state)
                     return
+
+    @provide_session
+    def execute_complete(self, context: Context, session: "Session", **kwargs):
+        parsed_execution_date = _parse_execution_date(self.execution_date)
+
+        dag_run = (
+            session.query(DagRun)
+            .filter(DagRun.dag_id == self.trigger_dag_id, DagRun.execution_date == parsed_execution_date)
+            .all()
+        )
+
+        if len(dag_run) > 1:
+            raise AirflowException(
+                f"{self.trigger_dag_id} with execution_data of {self.execution_date} appears too"
+                " many times, {len(dag_run)}"
+            )
+
+        dag_run = dag_run[0]
+
+        if not dag_run:
+            raise AirflowException(
+                f"{self.trigger_dag_id} with execution_data of {self.execution_date} doesn't"
+                " appear to exist"
```

Comment:
```suggestion
                No DAG run found for DAG f"{self.trigger_dag_id} and execution date {self.execution_date}"
```

Would make the message more explicit.